tests/test_integration.py

```diff
@@ -37,38 +37,47 @@ def passive_node():
 
 
 @asynccontextmanager
-async def create_channel(src, dest, funding: int):
+async def create_channel(src, dest, funding: int, close_from_dest=False):
     channel = await src["api"].open_channel(dest["address"], str(int(funding)))
     assert channel is not None
     await asyncio.wait_for(check_channel_status(src, dest, status="Open"), 10.0)
     try:
         yield channel
     finally:
-        assert await src["api"].close_channel(channel)
-        await asyncio.wait_for(check_channel_status(src, dest, status="PendingToClose"), 10.0)
+        if close_from_dest:
+            assert await dest["api"].close_channel(channel)
+            await asyncio.wait_for(check_channel_status(src, dest, status="Closed"), 10.0)
+        else:
+            assert await src["api"].close_channel(channel)
+            await asyncio.wait_for(check_channel_status(src, dest, status="PendingToClose"), 10.0)
```

Comment:
This duality of immediate and postponed closure is strange from the API perspective.

Reply:
I agree, we can split up later if you want to.